### PR TITLE
Backport: fix semi-sync for Backup and RestoreFromBackup commands by calling SetReplicationSource RPC

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

Backport PR for #9725 
After this change, vtctld will send an additional RPC for `SetReplicationSource` after `Backup` and `RestoreFromBackup` have run. This is needed for maintaining backward compatibility during the upgrade downgrade process.

vtctld from (version 13 and 14) - Will pass the correct semi-sync information in the `SetReplicationSource` RPC
vttablets from version 13 - Will ignore the semi-sync value passed and will use the `enable_semi_sync` flag
vttablets from version 14 - Will use the information passed in the RPC to set the semi-sync correctly.

vttablets from version 13 could have made autonomous decisions on whether semi-sync is supposed to be set or not. But in release 14, `enable_semi_sync` is deprecated, and the only way to know whether to set semi-sync or not will come from durability policies.

Without this backport, vtctld v13 will not issue the SetReplicationSource rpc and vttablets from v14 will not be able to fix semi-sync on its own, so Backups would have incorrect semi-sync settings during the upgrade process. This back port fixes this issue 

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/wrangler/testlib/backup_test.go [hunk 1]
```diff
@@ -684,8 +702,10 @@ func TestDisableActiveReparents(t *testing.T) {
 			},
 		},
 	}
+	sourceTablet.FakeMysqlDaemon.SetReplicationSourceInputs = []string{fmt.Sprintf("%s:%d", primary.Tablet.MysqlHostname, primary.Tablet.MysqlPort)}
 	sourceTablet.FakeMysqlDaemon.ExpectedExecuteSuperQueryList = []string{
 		"STOP SLAVE",
+		"FAKE SET MASTER",
```
Comment (deepthi) [inline go/vt/wrangler/testlib/backup_test.go:708]: This looks wrong.. when `DisableActiveReparents` is set, we _don't_ want to start replication.
What am I missing?

Comment (GuptaManan100) [reply go/vt/wrangler/testlib/backup_test.go:708]: We aren't starting replication, we are just setting the correct primary

Comment (deepthi) [reply go/vt/wrangler/testlib/backup_test.go:708]: Let me rephrase. We should not touch replication in _ANY_ way when this flag is set.

Comment (GuptaManan100) [reply go/vt/wrangler/testlib/backup_test.go]: Fixed

File: go/vt/wrangler/tablet.go [hunk 2]
```diff
@@ -127,30 +127,48 @@ func (wr *Wrangler) StartReplication(ctx context.Context, tablet *topodatapb.Tab
 	return wr.TabletManagerClient().StartReplication(ctx, tablet, semiSync)
 }
 
+// SetReplicationSource is used to set the replication source on the specified tablet
```
Comment (deepthi) [inline go/vt/wrangler/tablet.go:130]:
```suggestion
// SetReplicationSource is used to set the replication source on the specified tablet to the current shard primary.
```

File: go/vt/wrangler/tablet.go [hunk 3]
```diff
@@ -127,30 +127,48 @@ func (wr *Wrangler) StartReplication(ctx context.Context, tablet *topodatapb.Tab
...
+// It also finds out if the tablet should be sending semi-sync ACKs or not. It does not start the replication forcefully
```
Comment (deepthi) [inline go/vt/wrangler/tablet.go:131]:
```suggestion
// It also figures out if the tablet should be sending semi-sync ACKs or not and passes that to the tabletmanager RPC.
// It does not start replication forcefully.
```

**Review by deepthi — APPROVED** [review]
Looks good to me.
Can be merged once comments are fixed, and tests are passing.

File: go/vt/wrangler/testlib/planned_reparent_shard_test.go [hunk 4]
```diff
@@ -686,6 +689,10 @@ func TestPlannedReparentShardRelayLogErrorStartReplication(t *testing.T) {
 	// simulate error that will trigger a call to RestartReplication
 	goodReplica1.FakeMysqlDaemon.StartReplicationError = errors.New("Slave failed to initialize relay log info structure from the repository")
 	goodReplica1.FakeMysqlDaemon.ExpectedExecuteSuperQueryList = []string{
+		// In SetReplicationSource, we find that the source host and port was already set correctly,
+		// So we try to stop and start replication. The first STOP SLAVE comes from there
+		"STOP SLAVE",
+		// During the START SLAVE call, we find a relay log error, so we try to restart replication.
```
Comment (deepthi) [inline go/vt/wrangler/testlib/planned_reparent_shard_test.go:695]: 👍
